codalab/apps/chahub/utils.py

```diff
@@ -0,0 +1,74 @@
+from django.conf import settings
+
+import logging
+import os
+import requests
+import json
+
+logger = logging.getLogger(__name__)
+
+
+def send_to_chahub(endpoint, raw_data):
+    """
+    Does a post request to the specified API endpoint on chahub with the inputted data.
+    :param endpoint: String designating which API endpoint; IE: 'producers/'
+    :param raw_data: Dictionary containing data we are sending away to the endpoint.
+    :return:
+    """
+    try:
+        _chahub_api_url_check()
+    except KeyError:
+        logger.info("CHAHUB_API_URL not defined in settings!")
+        return
+
+    chahub_online = _chahub_online_check()
+
+    if not chahub_online:
```

Comment:
I don't think we need the temp variable here, can rewrite this function probably like...

```python
def send_to_chahub(endpoint, data):
    assert not settings.CHAHUB_API_URL, "CHAHUB_API_URL not defined in settings!"
    if not _chahub_is_online():
        logger.info("Chahub is currently offline. Cancelling.")
        raise requests.ConnectionError("Chahub is currently offline. Please try again later.")

    url = "{}{}".format(settings.CHAHUB_API_URL, endpoint)
    _chahub_send_data(url, json.dumps(data))
```

`raw_data` should probably just be `data`? We also handle the connection error stuff too many times -- in fact I think that would block the exception from being raised, and no retries would fire

Reply:
Also, having this function and `_chahub_send_data()` doesn't make a ton of sense, I think? should probably move the contents of that functions into this one?